### DISCUSSION TITLE
Does anyone have a working V2 camera

### DISCUSSION THREAD

**Discussion by bjs-pdx** [gtxaspec/wz_mini_hacks#475] [Q&A] [answered]
I'm struggling to get a V2 camera working with wz_mini and the Uctronics POE adapter.  I've tried with an older version and the current version of the hack.  I can get it to load, can access the web interface and ssh before I try to switch to ethernet, but the moment I activate ethernet in the configuration and reboot the camera cannot get an IP address and goes offline.  I'm attaching the Uctronics adapter using a USB A to micro adapter and this configuration works with my laptop so I know that everything is okay from a hardware perspective.

I'm using firmware version 4.9.8.1002 which I downloaded to the V2 install directory and then run the compile script.  I can see the camera in the Wyze App and can confirm it is running that firmware.  I hear the announcement from the camera on boot-up regarding the loading of the hacks and extraction of the swap file.

So, I need to ask, does anyone have a working V2 with ANY ethernet adapter?  Any with the Uctronics?

**Comment by bjs-pdx** [comment]
Here is my setup with a Black V2:

![IMG_0316](https://user-images.githubusercontent.com/14082937/217386857-0706ccbf-d539-415c-86b2-759779886380.jpg)

![IMG_0315](https://user-images.githubusercontent.com/14082937/217386897-4695d5ec-30c6-4b35-9427-bd5bcf7cb222.jpg)

![IMG_0314](https://user-images.githubusercontent.com/14082937/217386925-bf63332f-98f0-4178-91de-50b67c18acf3.jpg)

**Reply by Bob-AOL** [reply]
Maybe a careful read-through here will help. #241

Don't I recall you having the black one working some time ago in another thread somewhere?

**Reply by bjs-pdx** [reply]
I thought I had one working, but can't remember what I did.  Had knee replacement surgery in September so other things are kind of a blur and less important, all things considered..

**Comment by bjs-pdx** [comment]
@gtxaspec

Do you have any suggestions for troubleshooting the non-workng Uctronics adapter on a V2?

**Reply by Bob-AOL** [reply]
May want to revisit #250

**Reply by bjs-pdx** [reply]
Not any help there.  The problem is the V2 boots just fine from the altered firmware.  You can get to the web interface and ssh when it is connected to the wifi.  Then, when you activate ethernet, all connections are lost and there isn't any network connection.  Same problem as issue https://github.com/gtxaspec/wz_mini_hacks/issues/424

Looking for other ways to troubleshoot the problem.  Trying to find out if anyone has a working V2 and which NIC is being used.  So far only wifi.

**Reply by Bob-AOL** [reply]
#424 addresses T-31 devices. Different animals. Different problems. Different solutions.

> Looking for other ways to troubleshoot the problem.

Logs?

And Wireshark could certainly determine if there's any attempt by wz_mini to initiate dhcp discover. If there's nothing there, you're back to the compatibility issue I was facing with my two problem v2s.

Have you tried enabling default bonding? There was a caveat there somewhere as I recall.

The last problem I had with similar symptoms ended up being an SD card that played nice on all but one of my v3s. Got a couple new cheapies from Micro Center already formatted, loaded one up with wz_mini, popped it in the cam, worked as intended without a hitch and has been solid since.

I also recall we were pretty much on our own with the v2s last time around. Looks like you're pretty much the last man standing now.

Gotta be something simple being overlooked. You had them working with UCTRONICS adapters. I'm still searching for the thread...

**Comment by bjs-pdx** [comment, marked as answer]
From the log on the v2:

#####S03ethernet#####
Auto-Detect an Ethernet Driver and load it
Loading Realtek RTL8152 driver...
modprobe: module 'r8152' not found
Manually load any other Ethernet Drivers if asked for
cat: can't open '/sys/class/net/eth0/address': No such file or directory


Looks like the ethernet driver doesn't see the card.  Same problem as the other thread.  I've used different cameras, different sd cards and different Uctronics adapters and all produce the same symptoms and failure to initialize.  But the OS does see the adapter:

[root@Thing1:log]# lsusb
Bus 001 Device 002: ID 0bda:8152
Bus 001 Device 001: ID 1d6b:0002
[root@Thing1:log]#

**Reply by bjs-pdx** [reply]
Now I'm going crazy.  Setup the V2 with wifi and wz_mini and connected it to my network.  Works fine.  Connected via SSH, modified wz_mini.conf file for the following:  auto load false, manual load r8152, bonding true.  Then I rebooted the camera and it connected with a new MAC and I could see the camera in the app, connect via the web interface.  I then used SSH to connect, set bonding to false and rebooted and the camera is still connected using the Uctronics adapter.  Not sure why this worked...maybe the manual load (which I tried before) or something about having bonding enabled.  Rebooted it a couple of times and it still connects so I have no idea why it worked now.

![Screenshot 2023-02-15 at 15-54-02 T20(WYZEC1-JZ) on 9f242faa1f running wz_mini Thu Oct 13 12 37 18 AM PDT 2022 as Thing1](https://user-images.githubusercontent.com/14082937/219221972-d1830e1c-5905-4dc6-a1b3-18939b17b22a.png)

![IMG_0761](https://user-images.githubusercontent.com/14082937/219221718-24723ee6-1be9-4b2e-94b3-2092a39bab80.jpg)

**Reply by bjs-pdx** [reply]
One more bit of strangeness...can't get time to sync.  Off by about 20 minutes and won't sync using the app.  Added a time server to the config (time.google.com) and now it seems happy.

**Reply by Bob-AOL** [reply]
Cool. Uhhh..., congrats? I knew there was something with the bonding that worked even though it didn't make sense. 

Now tell the truth..., was it worth all the hassle? LOL

**Reply by bjs-pdx** [reply]
Not sure if it was worth it or not.  But I don't like to lose, especially to technology.  I have one spot for this camera and having POE would be very useful as it is at the very edge of my wifi.  Still not sure why it suddenly started working.  The MAC didn't change when I switched from bonding to no bonding.  I checked the MAC (wlan0     Link encap:Ethernet  HWaddr 00:E0:4C:xx:xx:xx)  and it comes up as RealTek so for the moment I'm happy.

**Reply by bjs-pdx** [reply]
Got the second V2 working.  One problem I have seems to be the USB A adapter I was trying. Seems it wasn't passing data between the camera and Uctronics adapter.  When I used lsusb to list the attached USB devices the Uctronics wasn't in the list.  Switched to a different USB A adapter and it started working immediately.
